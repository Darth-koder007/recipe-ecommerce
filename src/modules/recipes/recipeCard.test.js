import { fireEvent, render, screen } from '@testing-library/react';

import React from 'react';
import RecipeCard from './RecipeCard';
import recipeCardFixtures from './recipeCard.fixtures';

describe('Recipe card', () => {
  describe('default view', () => {
    test('should render recipe card with correct recipe name', () => {
      render(<RecipeCard {...recipeCardFixtures.testDefaultData} />);

      expect(screen.getByText(recipeCardFixtures.testDefaultData.name)).toBeVisible();
    });

    test('should render recipe card with correct recipe image', () => {
      render(<RecipeCard {...recipeCardFixtures.testDefaultData} />);

      expect(screen.getByAltText(recipeCardFixtures.testDefaultData.name)).toBeVisible();
    });

    test('should render recipe card with correct headline', () => {
      render(<RecipeCard {...recipeCardFixtures.testDefaultData} />);

      expect(screen.getByText(recipeCardFixtures.testDefaultData.headline)).toBeVisible();
    });

    test('should render recipe card with Add button', () => {
      render(<RecipeCard {...recipeCardFixtures.testDefaultData} />);

      const addButton = screen.getByRole('button', {
        name: `Add ${recipeCardFixtures.testDefaultData.name} to your hello fresh box`,
      });

      expect(addButton).toBeVisible();
    });

    test('should call handleAddRecipe function when clicked on Add button with correct recipe id', () => {
      render(<RecipeCard {...recipeCardFixtures.testDefaultData} />);

      const addButton = screen.getByRole('button', {
        name: `Add ${recipeCardFixtures.testDefaultData.name} to your hello fresh box`,
      });

      fireEvent.click(addButton);

      expect(recipeCardFixtures.testDefaultData.handleAddRecipe).toHaveBeenCalledWith(
        recipeCardFixtures.testDefaultData.id
      );
    });

    test('should render recipe card with extra charge text', () => {
      render(<RecipeCard {...recipeCardFixtures.testDefaultData} />);

      expect(screen.getByLabelText('Extra charge')).toBeVisible();
    });

    test('should render recipe card with Add button disabled when maximum recipe limit is reached', () => {
      render(<RecipeCard {...recipeCardFixtures.testAddDisabledData} />);

      const addButton = screen.getByRole('button', {
        name: `Add ${recipeCardFixtures.testAddDisabledData.name} to your hello fresh box`,
      });

      expect(addButton).toBeDisabled();
    });

    test('should match snapshot correctly with Add button', () => {
      expect(render(<RecipeCard {...recipeCardFixtures.testDefaultData} />)).toMatchSnapshot();
    });
  });

  describe('add extra meal view', () => {
    test('should render recipe card with Add extra meal button when minimum recipe limit is met', () => {
      render(<RecipeCard {...recipeCardFixtures.testAddExtraData} />);

      const addButton = screen.getByRole('button', {
        text: `Add ${recipeCardFixtures.testAddExtraData.name} to your hello fresh box`,
      });

      expect(addButton).toHaveTextContent('Add extra meal');
    });

    test('should render recipe card with Add extra meal button disabled when maximum recipe limit is reached', () => {
      render(<RecipeCard {...recipeCardFixtures.testAddExtraDisabledData} />);

      const addButton = screen.getByRole('button', {
        text: `Add ${recipeCardFixtures.testAddExtraDisabledData.name} to your hello fresh box`,
      });

      expect(addButton).toBeDisabled();
    });

    test('should match snapshot correctly with Add extra meal button', () => {
      expect(render(<RecipeCard {...recipeCardFixtures.testAddExtraData} />)).toMatchSnapshot();
    });
  });

  describe('quantity button view', () => {
    test('should render recipe card with selected quantity text', () => {
      render(<RecipeCard {...recipeCardFixtures.testQuantityButtonData} />);

      expect(
        screen.getByText(`${recipeCardFixtures.testQuantityButtonData.selected} in your box`)
      ).toBeVisible();
    });

    test('should render recipe card with increase and decrease quantity button', () => {
      expect.assertions(2);

      render(<RecipeCard {...recipeCardFixtures.testQuantityButtonData} />);

      const increaseButton = screen.getByRole('button', {
        name: `Increase quantity of ${recipeCardFixtures.testQuantityButtonData.name}`,
      });

      expect(increaseButton).toBeVisible();

      const decreaseButton = screen.getByRole('button', {
        name: `Decrease quantity of ${recipeCardFixtures.testQuantityButtonData.name}`,
      });

      expect(decreaseButton).toBeVisible();
    });

    test('should render recipe card with increase quantity button disabled when maximum recipe limit is reached', () => {
      render(<RecipeCard {...recipeCardFixtures.testQuantityButtonDisabledData} />);

      const increaseButton = screen.getByRole('button', {
        name: `Increase quantity of ${recipeCardFixtures.testQuantityButtonDisabledData.name}`,
      });

      expect(increaseButton).toBeDisabled();
    });

    test('should call handleAddRecipe function with correct recipe id when clicked on increase quantity button', () => {
      render(<RecipeCard {...recipeCardFixtures.testQuantityButtonData} />);

      const increaseButton = screen.getByRole('button', {
        name: `Increase quantity of ${recipeCardFixtures.testQuantityButtonData.name}`,
      });

      fireEvent.click(increaseButton);

      expect(recipeCardFixtures.testQuantityButtonData.handleAddRecipe).toHaveBeenCalledWith(
        recipeCardFixtures.testQuantityButtonData.id
      );
    });

    test('should call handleRemoveRecipe function with correct recipe id when clicked on descrease quantity button', () => {
      render(<RecipeCard {...recipeCardFixtures.testQuantityButtonData} />);

      const decreaseButton = screen.getByRole('button', {
        name: `Decrease quantity of ${recipeCardFixtures.testQuantityButtonData.name}`,
      });

      fireEvent.click(decreaseButton);

      expect(recipeCardFixtures.testQuantityButtonData.handleRemoveRecipe).toHaveBeenCalledWith(
        recipeCardFixtures.testQuantityButtonData.id
      );
    });

    test('should render recipe card with servings size text', () => {
      render(<RecipeCard {...recipeCardFixtures.testQuantityButtonData} />);

      expect(
        screen.getByText(
          `(${
            recipeCardFixtures.testQuantityButtonData.selected *
            recipeCardFixtures.testQuantityButtonData.yields
          } servings)`
        )
      ).toBeVisible();
    });

    test('should match snapshot correctly with increase and decrease quantity button', () => {
      expect(
        render(<RecipeCard {...recipeCardFixtures.testQuantityButtonData} />)
      ).toMatchSnapshot();
    });
  });
});
