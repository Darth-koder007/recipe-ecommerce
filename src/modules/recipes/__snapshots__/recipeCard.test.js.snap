// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Recipe card add extra meal view should match snapshot correctly with Add extra meal button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnylx gccBOx"
      >
        <div
          class="sc-bdnylx bZyySU"
          height="0"
          overflow="hidden"
        >
          <img
            alt="Steak & Shrimp in a Creamy Thyme Sauce"
            src="https://img.hellofresh.com/c_fill,f_auto,fl_lossy,h_405,q_auto,w_720/hellofresh_s3/image/steak-shrimp-in-a-creamy-thyme-sauce-12478c6a.jpg"
            width="100%"
          />
        </div>
        <div
          class="sc-bdnylx ksClEu"
          height="120px"
        >
          <div
            class="sc-eCApGN gsoVNo"
            font-family="primary"
            font-size="md"
            font-weight="bold"
          >
            Steak & Shrimp in a Creamy Thyme Sauce
          </div>
          <div
            class="sc-eCApGN kWxwPa"
            font-family="secondary"
            font-size="sm"
            font-weight="regular"
          >
            with Zesty Roasted Green Beans & Crispy-Onion-Topped Mashed Potatoes
          </div>
        </div>
        <div
          class="sc-bdnylx sc-dlnjPT jAJxLH"
          display="flex"
        >
          <div
            class="sc-bdnylx fqNRAk"
          >
            <div
              aria-label="Extra charge"
              class="sc-eCApGN kRAlbB"
              color="primary_600"
            >
              +
              $11.98
            </div>
          </div>
          <div
            class="sc-bdnylx hCkeme"
          >
            <button
              aria-label="Add Steak & Shrimp in a Creamy Thyme Sauce to your hello fresh box"
              class="sc-gtssRu kSeqjR"
              width="100%"
            >
              Add extra meal
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnylx gccBOx"
    >
      <div
        class="sc-bdnylx bZyySU"
        height="0"
        overflow="hidden"
      >
        <img
          alt="Steak & Shrimp in a Creamy Thyme Sauce"
          src="https://img.hellofresh.com/c_fill,f_auto,fl_lossy,h_405,q_auto,w_720/hellofresh_s3/image/steak-shrimp-in-a-creamy-thyme-sauce-12478c6a.jpg"
          width="100%"
        />
      </div>
      <div
        class="sc-bdnylx ksClEu"
        height="120px"
      >
        <div
          class="sc-eCApGN gsoVNo"
          font-family="primary"
          font-size="md"
          font-weight="bold"
        >
          Steak & Shrimp in a Creamy Thyme Sauce
        </div>
        <div
          class="sc-eCApGN kWxwPa"
          font-family="secondary"
          font-size="sm"
          font-weight="regular"
        >
          with Zesty Roasted Green Beans & Crispy-Onion-Topped Mashed Potatoes
        </div>
      </div>
      <div
        class="sc-bdnylx sc-dlnjPT jAJxLH"
        display="flex"
      >
        <div
          class="sc-bdnylx fqNRAk"
        >
          <div
            aria-label="Extra charge"
            class="sc-eCApGN kRAlbB"
            color="primary_600"
          >
            +
            $11.98
          </div>
        </div>
        <div
          class="sc-bdnylx hCkeme"
        >
          <button
            aria-label="Add Steak & Shrimp in a Creamy Thyme Sauce to your hello fresh box"
            class="sc-gtssRu kSeqjR"
            width="100%"
          >
            Add extra meal
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Recipe card default view should match snapshot correctly with Add button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnylx gccBOx"
      >
        <div
          class="sc-bdnylx bZyySU"
          height="0"
          overflow="hidden"
        >
          <img
            alt="Steak & Shrimp in a Creamy Thyme Sauce"
            src="https://img.hellofresh.com/c_fill,f_auto,fl_lossy,h_405,q_auto,w_720/hellofresh_s3/image/steak-shrimp-in-a-creamy-thyme-sauce-12478c6a.jpg"
            width="100%"
          />
        </div>
        <div
          class="sc-bdnylx ksClEu"
          height="120px"
        >
          <div
            class="sc-eCApGN gsoVNo"
            font-family="primary"
            font-size="md"
            font-weight="bold"
          >
            Steak & Shrimp in a Creamy Thyme Sauce
          </div>
          <div
            class="sc-eCApGN kWxwPa"
            font-family="secondary"
            font-size="sm"
            font-weight="regular"
          >
            with Zesty Roasted Green Beans & Crispy-Onion-Topped Mashed Potatoes
          </div>
        </div>
        <div
          class="sc-bdnylx sc-dlnjPT jAJxLH"
          display="flex"
        >
          <div
            class="sc-bdnylx fqNRAk"
          >
            <div
              aria-label="Extra charge"
              class="sc-eCApGN kRAlbB"
              color="primary_600"
            >
              +
              $11.98
            </div>
          </div>
          <div
            class="sc-bdnylx hCkeme"
          >
            <button
              aria-label="Add Steak & Shrimp in a Creamy Thyme Sauce to your hello fresh box"
              class="sc-gtssRu kSeqjR"
              width="100%"
            >
              Add
            </button>
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnylx gccBOx"
    >
      <div
        class="sc-bdnylx bZyySU"
        height="0"
        overflow="hidden"
      >
        <img
          alt="Steak & Shrimp in a Creamy Thyme Sauce"
          src="https://img.hellofresh.com/c_fill,f_auto,fl_lossy,h_405,q_auto,w_720/hellofresh_s3/image/steak-shrimp-in-a-creamy-thyme-sauce-12478c6a.jpg"
          width="100%"
        />
      </div>
      <div
        class="sc-bdnylx ksClEu"
        height="120px"
      >
        <div
          class="sc-eCApGN gsoVNo"
          font-family="primary"
          font-size="md"
          font-weight="bold"
        >
          Steak & Shrimp in a Creamy Thyme Sauce
        </div>
        <div
          class="sc-eCApGN kWxwPa"
          font-family="secondary"
          font-size="sm"
          font-weight="regular"
        >
          with Zesty Roasted Green Beans & Crispy-Onion-Topped Mashed Potatoes
        </div>
      </div>
      <div
        class="sc-bdnylx sc-dlnjPT jAJxLH"
        display="flex"
      >
        <div
          class="sc-bdnylx fqNRAk"
        >
          <div
            aria-label="Extra charge"
            class="sc-eCApGN kRAlbB"
            color="primary_600"
          >
            +
            $11.98
          </div>
        </div>
        <div
          class="sc-bdnylx hCkeme"
        >
          <button
            aria-label="Add Steak & Shrimp in a Creamy Thyme Sauce to your hello fresh box"
            class="sc-gtssRu kSeqjR"
            width="100%"
          >
            Add
          </button>
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Recipe card quantity button view should match snapshot correctly with increase and decrease quantity button 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="sc-bdnylx cDgClw"
      >
        <div
          class="sc-bdnylx bZyySU"
          height="0"
          overflow="hidden"
        >
          <img
            alt="Steak & Shrimp in a Creamy Thyme Sauce"
            src="https://img.hellofresh.com/c_fill,f_auto,fl_lossy,h_405,q_auto,w_720/hellofresh_s3/image/steak-shrimp-in-a-creamy-thyme-sauce-12478c6a.jpg"
            width="100%"
          />
        </div>
        <div
          class="sc-bdnylx ksClEu"
          height="120px"
        >
          <div
            class="sc-eCApGN gsoVNo"
            font-family="primary"
            font-size="md"
            font-weight="bold"
          >
            Steak & Shrimp in a Creamy Thyme Sauce
          </div>
          <div
            class="sc-eCApGN kWxwPa"
            font-family="secondary"
            font-size="sm"
            font-weight="regular"
          >
            with Zesty Roasted Green Beans & Crispy-Onion-Topped Mashed Potatoes
          </div>
        </div>
        <div
          class="sc-bdnylx sc-dlnjPT gKxRuq"
          display="flex"
        >
          <button
            aria-label="Decrease quantity of Steak & Shrimp in a Creamy Thyme Sauce"
            class="sc-jSFkmK eHNmvr"
            title="Decrease quantity of Steak & Shrimp in a Creamy Thyme Sauce"
          >
            <svg
              class="sc-hKFyIo ggweEv"
              height="24"
              viewBox="0 0 32 32"
              width="24"
            >
              <path
                d="M16 3c7.18 0 13 5.82 13 13s-5.82 13-13 13S3 23.18 3 16 8.82 3 16 3zm0 1.5C9.649 4.5 4.5 9.649 4.5 16S9.649 27.5 16 27.5 27.5 22.351 27.5 16 22.351 4.5 16 4.5zM22 15v2H10v-2h12z"
              />
            </svg>
          </button>
          <div
            class="sc-bdnylx kwhRwX"
            color="white"
          >
            <div
              class="sc-eCApGN iBCCQJ"
              font-family="secondary"
              font-size="md"
              font-weight="bold"
            >
              3
               in your box
            </div>
            <div
              class="sc-eCApGN eoOXgE"
              font-family="secondary"
              font-size="sm"
            >
              (
              6
               servings)
            </div>
          </div>
          <button
            aria-label="Increase quantity of Steak & Shrimp in a Creamy Thyme Sauce"
            class="sc-jSFkmK eHNmvr"
            title="Increase quantity of Steak & Shrimp in a Creamy Thyme Sauce"
          >
            <svg
              class="sc-hKFyIo ggweEv"
              height="24"
              viewBox="0 0 32 32"
              width="24"
            >
              <path
                d="M16 3c7.18 0 13 5.82 13 13s-5.82 13-13 13S3 23.18 3 16 8.82 3 16 3zm0 1.5C9.649 4.5 4.5 9.649 4.5 16S9.649 27.5 16 27.5 27.5 22.351 27.5 16 22.351 4.5 16 4.5zm1 5.5v5h5v2h-5v5h-2v-5h-5v-2h5v-5h2z"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="sc-bdnylx cDgClw"
    >
      <div
        class="sc-bdnylx bZyySU"
        height="0"
        overflow="hidden"
      >
        <img
          alt="Steak & Shrimp in a Creamy Thyme Sauce"
          src="https://img.hellofresh.com/c_fill,f_auto,fl_lossy,h_405,q_auto,w_720/hellofresh_s3/image/steak-shrimp-in-a-creamy-thyme-sauce-12478c6a.jpg"
          width="100%"
        />
      </div>
      <div
        class="sc-bdnylx ksClEu"
        height="120px"
      >
        <div
          class="sc-eCApGN gsoVNo"
          font-family="primary"
          font-size="md"
          font-weight="bold"
        >
          Steak & Shrimp in a Creamy Thyme Sauce
        </div>
        <div
          class="sc-eCApGN kWxwPa"
          font-family="secondary"
          font-size="sm"
          font-weight="regular"
        >
          with Zesty Roasted Green Beans & Crispy-Onion-Topped Mashed Potatoes
        </div>
      </div>
      <div
        class="sc-bdnylx sc-dlnjPT gKxRuq"
        display="flex"
      >
        <button
          aria-label="Decrease quantity of Steak & Shrimp in a Creamy Thyme Sauce"
          class="sc-jSFkmK eHNmvr"
          title="Decrease quantity of Steak & Shrimp in a Creamy Thyme Sauce"
        >
          <svg
            class="sc-hKFyIo ggweEv"
            height="24"
            viewBox="0 0 32 32"
            width="24"
          >
            <path
              d="M16 3c7.18 0 13 5.82 13 13s-5.82 13-13 13S3 23.18 3 16 8.82 3 16 3zm0 1.5C9.649 4.5 4.5 9.649 4.5 16S9.649 27.5 16 27.5 27.5 22.351 27.5 16 22.351 4.5 16 4.5zM22 15v2H10v-2h12z"
            />
          </svg>
        </button>
        <div
          class="sc-bdnylx kwhRwX"
          color="white"
        >
          <div
            class="sc-eCApGN iBCCQJ"
            font-family="secondary"
            font-size="md"
            font-weight="bold"
          >
            3
             in your box
          </div>
          <div
            class="sc-eCApGN eoOXgE"
            font-family="secondary"
            font-size="sm"
          >
            (
            6
             servings)
          </div>
        </div>
        <button
          aria-label="Increase quantity of Steak & Shrimp in a Creamy Thyme Sauce"
          class="sc-jSFkmK eHNmvr"
          title="Increase quantity of Steak & Shrimp in a Creamy Thyme Sauce"
        >
          <svg
            class="sc-hKFyIo ggweEv"
            height="24"
            viewBox="0 0 32 32"
            width="24"
          >
            <path
              d="M16 3c7.18 0 13 5.82 13 13s-5.82 13-13 13S3 23.18 3 16 8.82 3 16 3zm0 1.5C9.649 4.5 4.5 9.649 4.5 16S9.649 27.5 16 27.5 27.5 22.351 27.5 16 22.351 4.5 16 4.5zm1 5.5v5h5v2h-5v5h-2v-5h-5v-2h5v-5h2z"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
